sympy/core/add.py

```diff
@@ -836,10 +836,31 @@ def _eval_as_leading_term(self, x):
 
         infinite = [t for t in expr.args if t.is_infinite]
 
-        expr = expr.func(*[t.as_leading_term(x) for t in expr.args]).removeO()
-        if not expr:
-            # simple leading term analysis gave us 0 but we have to send
-            # back a term, so compute the leading term (via series)
+        compute = False
+        leading_term = [t.as_leading_term(x) for t in expr.args]
+        new_expr = expr.func(*leading_term).removeO()
+
+        if not new_expr:
+            compute = True
+        elif new_expr is not S.NaN:
+            if new_expr.is_Add:
+                final_leading_term = [t for t in new_expr.args]
+            else:
+                final_leading_term = [new_expr]
+
+            if len(expr.args) != len(final_leading_term):
+                coeff_added_term = [t for t in leading_term if t not in final_leading_term]
+                leading_coeff = [t.as_coeff_Mul()[1] for t in final_leading_term]
```

Comment:
This will extract numeric coefficients but not symbolic ones.
```
>>> (3*x).as_coeff_Mul()
(3, x)
>>> (a*x).as_coeff_Mul()
(1, a*x)
```
Treating them differently will probably make the code more complicated.

Reply:
@jksuom  ```x+2*x=3*x```
**but**
 
```
>>> from sympy import *
>>> a=Symbol('a')
>>> x=Symbol('x')
>>> a*x+2*x
a*x + 2*x
>>> b=Symbol('b')
>>> a*x+b*x
a*x + b*x
```
So symbolic term does not add up and form a new term like ```x+2*x``` does.
```
>>> a*x+a*x
2*a*x
>>> a*x+a*x+2*a*x
4*a*x
>>> a*x-a*x
0
```
For this type of addition this code will work
```
>>> e=a/(1-x)-a+sin(a*x)+a*x
>>> e.as_leading_term(x)
3*a*x
>>> e=a/(1-x)-a+sin(a*x)+x
>>> e.as_leading_term(x)
x*(2*a + 1)
>>> e=a/(1-x)-a+sin(a*x)+b*x
>>> e.as_leading_term(x)
x*(2*a + b)
```
in that case this code will treat  ```a*x``` as a whole

Reply:
I find the name `leading_coeff` confusing since `t.as_coeff_Mul()` returns a tuple whose first component is the coefficient but that is ignored and only the second component is taken. Besides, a plural form name would have been more appropriate as the list may generally have several elements.

Reply:
@jksuom yes ok I have corrected it